### ISSUE TITLE
[Bug][macOS]: Crash when updating recommendations collection

### ISSUE THREAD

**Issue by anne1pete** [gdavis/Mimir-Feedback#221]
### Are you sure this is a new issue?

- [X] I have confirmed this is a new issue

### macOS Version

Monterey 12.6.2

### Your Mac

MacBook Air (M1, 2020)

### App Version

Mimir 1.5.1 (130)

### What happened?

Grant
First, I love this app!

I think a feed update issue

When I open Mimir after about 10 sec there is an update on my feed.  All goes well.
Later an automatic feed update begins and then crashes.
I wait for the Mac to give me a “report sent, do you want to reopen” window and I reopen with no problem. 
Been happening for a few months.  It occurred to me that apple may not be communicating my issue to you.

Attached are five copies of their reports that happened recently (Crash #3 is missing I accidentally deleted it).

Anne

[Crash 1.docx](https://github.com/gdavis/Mimir-Feedback/files/10317541/Crash.1.docx)
[Crash 2.docx](https://github.com/gdavis/Mimir-Feedback/files/10317542/Crash.2.docx)
[Crash 4.docx](https://github.com/gdavis/Mimir-Feedback/files/10317543/Crash.4.docx)
[Crash 5.docx](https://github.com/gdavis/Mimir-Feedback/files/10317544/Crash.5.docx)
[Crash 6.docx](https://github.com/gdavis/Mimir-Feedback/files/10317545/Crash.6.docx)

### Bug prevalence

Every time i use Mimir

### How does it happen?

When I open Mimir after about 10 sec there is an update on my feed.  All goes well.
Later an automatic feed update begins and then crashes.
I wait for the Mac to give me a “report sent, do you want to reopen” window and I reopen with no problem. 

### Add screenshots or video

Didn't do this but I can if you need it.

**Comment by gdavis** [comment]
Hello @anne1pete! Thank you for taking the time to report this. I've looked through the crash logs and believe I have found both the cause and solution for this crash. It seems to be a bug with how updates are happening when it tries to refresh the recommendations in the app. 

I'll be testing this fix out over the next few days and hope to have it live soon with another big update to the app I've been working on for some time.

Are you still able to use the app after the crash occurs once, or is this preventing you from using the app at all?
